L1Trigger/L1THGCal/plugins/backend/HGCalBackendStage1Processor.cc

```diff
@@ -36,14 +36,15 @@ class HGCalBackendStage1Processor : public HGCalBackendLayer1ProcessorBase {
       clustering_->eventSetup(es);
     if (clusteringDummy_)
       clusteringDummy_->eventSetup(es);
-   const l1t::HGCalTriggerCellBxCollection& collInput = *collHandle; 
+    const l1t::HGCalTriggerCellBxCollection& collInput = *collHandle; 
     /*To split jobs in fpga modules*/
-    std::unordered_map<uint32_t,std::vector<l1t::HGCalTriggerCell>> triggerCellsPtrs;
+ //   std::unordered_map<uint32_t,std::vector<l1t::HGCalTriggerCell>> triggerCellsPtrs;
+    std::unordered_map<uint32_t,std::vector<edm::Ptr<l1t::HGCalTriggerCell>>> triggerCellsPtrs;
 //    std::vector<edm::Ptr<l1t::HGCalTriggerCell> > triggerCellsPtrs;
     for (const auto& trigMod : collInput) {
       uint32_t module = geometry_->getModuleFromTriggerCell(trigMod.detId());
-      uint32_t fpga = geometry_->getStage1FpgaFromModule(trigMod.moduleId());
-      triggerCellsPtrs[module].push_back(trigMod);
+      uint32_t fpga = geometry_->getStage1FpgaFromModule(module);
+      triggerCellsPtrs[fpga].push_back(trigMod);
```

Comment:
`edm:Ptr` cannot be built directly from `HGCalTriggerCell` as they need some kind of pointing information to the original collection. The goal of using `edm:Ptr` here is to be able to link `HGCalTriggerCells` into `HGCalClusters` and `HGCalMultiClusters`.
The way to build an `edm:Ptr` of trigger cells is what was originally done here:
```cpp
    std::vector<edm::Ptr<l1t::HGCalTriggerCell>> triggerCellsPtrs;
    for (unsigned i = 0; i < collHandle->size(); ++i) {
      edm::Ptr<l1t::HGCalTriggerCell> ptr(collHandle, i);
      triggerCellsPtrs.push_back(ptr);
    }
```